dashboard/client/src/pages/node/NodeRow.tsx

```diff
@@ -0,0 +1,272 @@
+  const coreWorker = coreWorkerStats.length ? coreWorkerStats[0] : undefined;
+  const workerLogUrl = coreWorker
+    ? `/log/${encodeURIComponent(logUrl)}?fileName=${coreWorker.workerId}`
+    : `/log/${encodeURIComponent(logUrl)}`;
+
+  return (
+    <TableRow key={pid}>
+      <TableCell></TableCell>
+      <TableCell>
+        <StatusChip type="worker" status="ALIVE" />
+      </TableCell>
+      <TableCell align="center">
+        {coreWorker && (
+          <Tooltip title={coreWorker.workerId} arrow interactive>
+            <span className={classes.idCol}>{coreWorker.workerId}</span>
+          </Tooltip>
+        )}
+      </TableCell>
+      <TableCell align="center">{cmdline[0]}</TableCell>
+      <TableCell align="center">{pid}</TableCell>
+      <TableCell>
+        <PercentageBar num={Number(cpu)} total={100}>
+          {cpu}%
+        </PercentageBar>
+      </TableCell>
+      <TableCell>
+        <PercentageBar num={memoryInfo.rss} total={mem[0]}>
+          {memoryConverter(memoryInfo.rss)}/{memoryConverter(mem[0])}(
+          {(memoryInfo.rss / mem[0]).toFixed(1)}
+          %)
+        </PercentageBar>
+      </TableCell>
+      <TableCell>N/A</TableCell>
+      <TableCell>N/A</TableCell>
+      <TableCell align="center">N/A</TableCell>
+      <TableCell align="center">N/A</TableCell>
+      <TableCell>
+        <Link to={workerLogUrl} target="_blank">
+          Log
+        </Link>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+type NodeRowsProps = {
+  /**
+   * Details of the node
+   */
+  node: NodeDetail;
+  /**
+   * The index of the table. Needed to guarantee key uniqueness.
+   */
+  rowIndex: number;
+  /**
+   * Whether the node row should refresh data about its workers.
+   */
+  isRefreshing: boolean;
+};
+
+/**
+ * The rows related to a node and its workers. Expandable to show information about workers.
+ */
+export const NodeRows = ({ node, rowIndex, isRefreshing }: NodeRowsProps) => {
+  const [isExpanded, setExpanded] = useState(false);
+  const [workers, setWorkers] = useState<Worker[]>([]);
+  const tot = useRef<NodeJS.Timeout>();
+
+  const getDetail = useCallback(async () => {
+    if (!isRefreshing || !isExpanded) {
+      return;
+    }
+    const { data } = await getNodeDetail(node.raylet.nodeId);
+    const { data: rspData, result } = data;
+    if (rspData?.detail) {
+      const sortedWorkers = sortBy(
+        rspData.detail.workers,
+        (worker) => worker.pid,
+      );
+      setWorkers(sortedWorkers);
+    }
+
+    if (result === false) {
+      console.error("Node Query Error Please Check Node Name");
+    }
+
+    tot.current = setTimeout(getDetail, 4000);
+  }, [isRefreshing, isExpanded, node.raylet.nodeId]);
+
+  useEffect(() => {
+    getDetail();
+    return () => {
+      if (tot.current) {
+        clearTimeout(tot.current);
+      }
+    };
+  }, [getDetail]);
+
+  const handleExpandButtonClick = () => {
+    setExpanded(!isExpanded);
+  };
+
+  return (
+    <React.Fragment>
```

Comment:
use div as NodeRows's return instead of 'React.Fragment', if do that NodeRows and NodeRow do not need 'rowIndex'.

Reply:
i get you, and i know it's worked well here. maybe it has a better way but I don't know what to do right now. so just fine 😊